src/main/java/org/codinjutsu/tools/jenkins/view/util/BuildStatusIcon.java

```diff
@@ -48,44 +38,63 @@ public static JComponent createIcon(boolean combine, BuildStatusAggregator aggre
         JPanel combined = new JPanel();
 
         if (aggregator.hasNoResults()) {
-            return new BuildStatusIcon(buildStatusRenderer.renderBuildStatus(BuildStatusEnum.NULL),
-                    "No builds", 0);
+            return new BuildStatusIcon(buildStatusRenderer.renderBuildStatus(BuildStatusEnum.NULL), "No builds", 0);
         }
 
-        int nbBrokenBuilds = aggregator.getBrokenBuilds();
-        if (nbBrokenBuilds > 0) {
-            BuildStatusIcon brokenIcon = new BuildStatusIcon(buildStatusRenderer.renderBuildStatus(BuildStatusEnum.FAILURE),
-                    String.format("%d broken builds", nbBrokenBuilds), nbBrokenBuilds);
-            if (!combine) {
+        var brokenIcon = createIcon(aggregator.getBrokenBuilds(), buildStatusRenderer, BuildStatusEnum.FAILURE, "broken");
+        if (brokenIcon != null) {
+            if (combine) {
+                combined.add(brokenIcon);
+            } else {
                 return brokenIcon;
             }
-            combined.add(brokenIcon);
         }
 
-        int nbUnstableBuilds = aggregator.getUnstableBuilds();
-        if (nbUnstableBuilds > 0) {
-            BuildStatusIcon unstableIcon = new BuildStatusIcon(buildStatusRenderer.renderBuildStatus(BuildStatusEnum.UNSTABLE),
-                    String.format("%d unstable builds", nbUnstableBuilds), nbUnstableBuilds);
-            if (!combine) {
+        var unstableIcon = createIcon(aggregator.getUnstableBuilds(), buildStatusRenderer, BuildStatusEnum.UNSTABLE, "unstable");
+        if (unstableIcon != null) {
+            if (combine) {
+                combined.add(unstableIcon);
+            } else {
                 return unstableIcon;
             }
-            combined.add(unstableIcon);
         }
 
-        String succeededTooltip = combine ? String.format("%s succeeded builds", aggregator.getSucceededBuilds()) : "No broken builds";
-        int succeededNumber = combine ? aggregator.getSucceededBuilds() : 0;
-        BuildStatusIcon successIcon = new BuildStatusIcon(buildStatusRenderer.renderBuildStatus(BuildStatusEnum.SUCCESS), succeededTooltip, succeededNumber);
+        var runningIcon = createIcon(aggregator.getRunningBuilds(), buildStatusRenderer, BuildStatusEnum.RUNNING, "running");
+        if (runningIcon != null) {
+            if (combine) {
+                combined.add(runningIcon, 0);
+            } else {
+                return runningIcon;
+            }
+        }
+
+        var succeededIcon = createIcon(aggregator.getSucceededBuilds(), buildStatusRenderer, BuildStatusEnum.SUCCESS, "succeeded");
+        if (succeededIcon != null) {
+            if (combine) {
+                combined.add(succeededIcon);
+            } else {
+                return succeededIcon;
+            }
+        }
```

Comment:
My reasoning is as follows:
- Broken / Unstable -> Action is to fix the build
- Running -> Action is to keep an eye out on what it is going to do
- Successful / No builds -> No action needed

Because the 'running' status has a (passive) action and the 'successful' status has no action I think it's more important to show it. If successful has a higher priority then in many cases the running won't ever show up because there will hopefully always be at least one successful build :).